### PR TITLE
BUGFIX: site list in "content" of left drawer is policy checked

### PR DESCRIPTION
**Pull request by tdausner**
Using policies for `NodeTreePrivilege` (in `Policy.yaml`) did not reflect in the list of sites in left drawer "content" site list. The bugfix restricts user to only sees sites with `NodeTreePrivilege` access GRANTED.

Function `buildSiteList` is enriched by `$context` and `$privilegeManager`. For each site the node of path `/sites/{root-node-name-of-site}` is checked by `$privilegeManager->isGranted()` and put into site list on GRANTED.

**Use case / Review instructions**

- start with a fresh NEOS development installtion (see https://github.com/neos/neos-development-collection#contributing)
- kickstart multi site setup (First.Site, Second.Site, Third.Site) by command 
`$ for s in First Second Third; do ./flow kickstart:site $s.Site $s ; done`

- Import sites just created
`$ for s in First Second Third; do ./flow site:import --package-key $s.Site ; done`

- domain setup for sites in backed Administration/Sites Management
First.Site -> https://first.tds (_tds_ is my development domain)
Second.Site -> https://second.tds
Third.Site -> https://third.tds
image

- Setup `Policy.yaml` (see [Policy.yaml.txt](https://github.com/neos/neos-development-collection/files/10146777/Policy.yaml.txt) ) and check resulting roles by command
```
$ ./flow security:listroles
+-----------------------------------------------+-----------------------+
| Id                                            | Label                 |
+-----------------------------------------------+-----------------------+
| Neos.Neos:LivePublisher                       | Live publisher        |
| Neos.Neos:Administrator                       | Neos Administrator    |
| Neos.Neos:RestrictedEditor                    | Restricted Editor     |
| Neos.Neos:Editor                              | Editor                |
| Neos.Neos:UserManager                         | Neos User Manager     |
| Neos.RedirectHandler.Ui:RedirectAdministrator | RedirectAdministrator |
| Neos.Setup:SetupUser                          | SetupUser             |
| First.Site:FirstEditor                        | First site Editor     |
| Second.Site:SecondEditor                      | Second site Editor    |
| Third.Site:ThirdEditor                        | Third site Editor     |
+-----------------------------------------------+-----------------------+
```
`NodeTreePrivilege` matcher in `Policy.yaml` is `'isDescendantNodeOf("/sites/first-site")'` for First.Site and so on. Retrieve the node names by command:
```
$ ./flow site:list

 Name                 Node name                 Resources package               Status
----------------------------------------------------------------------------------------
 First                first-site                First.Site                      online
 Second               second-site               Second.Site                     online
 Third                third-site                Third.Site                      online

```
- Add test user:
`$ ./flow user:create --roles Second.Site:SecondEditor TestUser testuser-password Test User`

- Log into NEOS as administrator
image

- Log into NEOS as TestUser
image

Checked releases 5.0 and 7.0 source code. Also in theese releases regards to policies are missing. Looks like the bug is inherited.

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by tdausner** [comment]
@markusguenther lost my repo so the original pr got closed.

**Review by grebaldi — APPROVED** [review]
Hi @tdausner,

thanks to your excellent review instructions I was able to reproduce the problem and to verify manually that your change solves it :)

Code looks good to me :+1:

File: Neos.Neos/Classes/Controller/Backend/MenuHelper.php [hunk 1]
```diff
@@ -81,34 +89,38 @@ public function buildSiteList(ControllerContext $controllerContext): array
...
-                    return $domain->getHostname();
-                })->toArray();
-
-                $active = in_array($requestUriHost, $activeHostPatterns, true);
-
-                if ($active) {
-                    $uri = $contentModule['uri'];
-                } else {
-                    $uri = $controllerContext->getUriBuilder()->reset()->uriFor('switchSite', ['site' => $site], 'Backend\Backend', 'Neos.Neos');
+            $node = $context->getNode("/sites/" . $site->getNodeName());
```
Comment (bwaidelich) [inline Neos.Neos/Classes/Controller/Backend/MenuHelper.php]: Are site nodes _guaranteed_ to be underneath `/sites/`?
If so we should still replace the string with a constant (there is `SiteService::SITES_ROOT_PATH` IIRC) or at least replace the double quotes :)

File: Neos.Neos/Classes/Controller/Backend/MenuHelper.php [hunk 2]
```diff
@@ -58,6 +60,12 @@ class MenuHelper
      */
     protected $iconMapper;
 
+    /**
+     * @Flow\Inject
+     * @var \Neos\Neos\Domain\Service\ContentContextFactory
```
Comment (bwaidelich) [inline Neos.Neos/Classes/Controller/Backend/MenuHelper.php]: nitpick: This namespace could be imported for better readability

**Review by bwaidelich — COMMENTED** [review]
Looking good, by reading.
Just two tiny issues/questions

File: Neos.Neos/Classes/Controller/Backend/MenuHelper.php (hunk 1 above)
Comment (kdambekalns) [reply Neos.Neos/Classes/Controller/Backend/MenuHelper.php]: Yep, `Neos\Neos\Domain\Service\SiteService::SITES_ROOT_PATH` is set to `/sites`.
```suggestion
            $node = $context->getNode(\Neos\Neos\Domain\Service\SiteService::SITES_ROOT_PATH . '/' . $site->getNodeName());
```
(and please add an import for the `SiteService`)

Comment (tdausner) [reply Neos.Neos/Classes/Controller/Backend/MenuHelper.php]: @kdambekalns that's what I was looking for... done!

File: Neos.Neos/Classes/Controller/Backend/MenuHelper.php (hunk 2 above)
Comment (tdausner) [reply Neos.Neos/Classes/Controller/Backend/MenuHelper.php]: @bwaidelich All namespaces imported for better readability

**Comment by tdausner** [comment]
- '/sites' replaced by SiteService::SITES_ROOT_PATH
- use ...\SiteService; added
- missing property type declarations added / PhpDoc @throws updated

**Comment by kdambekalns** [comment]
Thanks! I was wondering… should this be fixed in 7.3, as it's a bugfix? I think yes…

If so, the type declarations won't work with injection. 😬

**Comment by tdausner** [comment]
Oooops... I'm so happy with Neos 8... didn't think about backwards compatibility. 
Removed type declarations on @Flow\Inject. 
Tested on
- Neos 7.3.10
- Neos 8.2 development collection

**Review by kdambekalns — APPROVED** [review]
Fine (by reading!)

**Comment by kdambekalns** [comment]
@tdausner Could you open this against 7.3 then?

**Comment by tdausner** [comment]
Obsolete by https://github.com/neos/neos-development-collection/pull/4025

**Pull request closed without merging**